rest/oidc_api_test.go

```diff
@@ -1,139 +1,455 @@
-		require.NotEmpty(t, location, "Location should be available in response header")
-		locationURL, err := url.Parse(location)
-		require.NoError(t, err, "Location header should be a valid URL")
-		redirectURI := locationURL.Query().Get(authRedirectURI)
-		require.NotEmpty(t, location, "redirect_uri should be available in auth URL")
-		redirectURL, err := url.Parse(redirectURI)
-		require.NoError(t, err, "redirect_uri should be a valid URL")
-		assert.Equal(t, "", redirectURL.Query().Get(authProvider))
-	})
+	type test struct {
+		name             string
+		inputRequestURL  string
+		wantProviderName string
+	}
+	tests := []test{
+		{
+			// When multiple providers are defined, default provider is specified and the current provider is
+			// not default, then current provider should be added to the generated OpenID Connect callback URL.
+			name:             "default_specified_but_current_is_not_default",
+			inputRequestURL:  "/db/_oidc?provider=facebook&offline=true",
+			wantProviderName: "facebook",
+		}, {
+			// When multiple providers are defined, default provider is specified and the current provider is
+			// default, then current provider should NOT be added to the generated OpenID Connect callback URL.
+			name:            "default_specified_and_current_is_default",
+			inputRequestURL: "/db/_oidc?provider=google&offline=true",
+		}, {
+			// When multiple providers are defined, default provider is specified and no current provider is
+			// provided, then provider name should NOT be added to the generated OpenID Connect callback URL.
+			name:            "default_specified_with_no_current",
+			inputRequestURL: "/db/_oidc?offline=true",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			resp := rt.SendAdminRequest(http.MethodGet, tc.inputRequestURL, "")
+			require.Equal(t, http.StatusFound, resp.Code)
+			location := resp.Header().Get(headerLocation)
+			require.NotEmpty(t, location, "Location should be available in response header")
+			locationURL, err := url.Parse(location)
+			require.NoError(t, err, "Location header should be a valid URL")
+			redirectURI := locationURL.Query().Get(requestParamRedirectURI)
+			require.NotEmpty(t, location, "redirect_uri should be available in auth URL")
+			redirectURL, err := url.Parse(redirectURI)
+			require.NoError(t, err, "redirect_uri should be a valid URL")
+			assert.Equal(t, tc.wantProviderName, redirectURL.Query().Get(auth.OIDCAuthProvider))
+		})
+	}
+}
+
+func TestCallbackState(t *testing.T) {
+	mockAuthServer, err := mockAuthServer()
+	require.NoError(t, err, "Error mocking fake authorization server")
+	defer mockAuthServer.Close()
+	issuerGoogle = mockAuthServer.URL + "/google"
+
+	type test struct {
+		name                   string
+		inputProviders         auth.OIDCProviderMap
+		inputDefaultProvider   string
+		inputRequestURL        string
+		inputChangeStateCookie bool
+		wantUsername           string
+		errorExpected          bool
```

Comment:
This should be more specific for what it's being used for

```suggestion
		invalidateState        bool
```

Reply:
Renamed errorExpected to invalidateState.